src/reactionsystem.jl

```diff
@@ -606,8 +606,8 @@ function make_ReactionSystem_internal(rxs::Vector{<:Reaction}, iv, no_sps::Nothi
             push!(ivs, value(siv))
         end
     end
-    sts = OrderedSet()
-    ps = OrderedSet{Any}(ps_in)
+    sts = OrderedSet{eltype(sts_in)}(sts_in)
```

Comment:
@torkel Does this seem reasonable to you instead of what you had with checking if sts_in is set equal to sts? It should still preserve the order of sts_in and seems simpler (and more in line with how we handle ps_in).

Reply:
I think this sounds good. The tests should check that it works as well.